### ISSUE TITLE
Fix statistics for sharded Solr cores

### ISSUE THREAD

**Issue by alanorth** [ilri/dspace-statistics-api#10]
A DSpace's Solr statistics cores may or may not be sharded depending on whether or not the site has run the [yearly `dspace stats-util -s` task](https://wiki.duraspace.org/display/DSDOC5x/Scheduled+Tasks+via+Cron). If the statistics core is sharded, views and download statistics returned by the API will be inaccurate because it currently indexes the `statistics` core.

To illustrate that this is problem see here the results of querying Solr for item views in `statistics` and `statistics-2018` cores for an item:

```
$ http 'http://localhost:3000/solr/statistics/select?indent=on&rows=0&q=type:2+id:11576&fq=isBot:false&fq=statistics_type:view' | grep numFound
<result name="response" numFound="33" start="0">
$ http 'http://localhost:3000/solr/statistics-2018/select?indent=on&rows=0&q=type:2+id:11576&fq=isBot:false&fq=statistics_type:view' | grep numFound
<result name="response" numFound="241" start="0">
```

So we definitely have to handle sharded cores somehow. [Perhaps with multicore join queries](https://wiki.apache.org/solr/Join).

**Comment by alanorth** [comment]
I discovered that I can [query multiple shards using a distributed search](https://wiki.apache.org/solr/DistributedSearch). For example:

```console
$ http 'http://localhost:8081/solr/statistics/select?indent=on&rows=0&q=type:2+id:11576&fq=isBot:false&fq=statistics_type:view&shards=localhost:8081/solr/statistics,localhost:8081/solr/statistics-2018' | grep numFound
<result name="response" numFound="274" start="0" maxScore="12.205825">
```

The result here is 274, which is 33 + 241 from the two queries above. I need to work this into the indexer component.

**Comment by alanorth** [comment]
Fixed in 40e284dac05bb3832b550bec9ef4d636c032b7dd.

**Comment by alanorth** [comment]
With all statistics shards being queried (2010–2018, plus the base `statistics` core) the number of item views is much higher:

```
$ http 'http://localhost:8081/solr/statistics/select?indent=on&rows=0&q=type:2+id:11576&fq=isBot:false&fq=statistics_type:view&shards=localhost:8081/solr/statistics,localhost:8081/solr/statistics-2018,localhost:8081/solr/statistics-2017,localhost:8081/solr/statistics-2016,localhost:8081/solr/statistics-2015,localhost:8081/solr/statistics-2014,localhost:8081/solr/statistics-2013,localhost:8081/solr/statistics-2012,localhost:8081/solr/statistics-2011,localhost:8081/solr/statistics-2010' | grep numFound
<result name="response" numFound="1775" start="0" maxScore="12.318736">
```
